nix_update/update.py

```diff
@@ -469,7 +476,28 @@ def update_version(
             package.diff_url = f"https://npmdiff.dev/{parts[1]}/{package.old_version}/{new_version.number}"
         elif package.parsed_url.netloc == "github.com":
             _, owner, repo, *_ = package.parsed_url.path.split("/")
-            package.diff_url = f"https://github.com/{owner}/{repo.removesuffix('.git')}/compare/{old_rev_tag}...{new_version.rev or new_version.number}"
+
+            if old_rev_tag is None:
+                # happens when using fetchurl with a github link rather than using fetchFromGitHub
+                regex = re.compile(".*/releases/download/(.*)/.*")
+                match = regex.match(package.parsed_url.path)
+                if match is not None:
+                    old_rev_tag = match.group(1)
+
+            new_rev_tag = new_version.rev or new_version.tag
+            if new_rev_tag is None:
+                # happens with fixed version preference (and possibly more situtations?)
+                new_package = eval_attr(opts)
+                new_rev_tag = new_package.rev or new_package.tag
+
+                if new_rev_tag is None and new_package.parsed_url is not None:
+                    # happens when using fetchurl with a github link rather than using fetchFromGitHub
+                    regex = re.compile(".*/releases/download/(.*)/.*")
+                    match = regex.match(new_package.parsed_url.path)
+                    if match is not None:
+                        new_rev_tag = match.group(1)
+
+            package.diff_url = f"https://github.com/{owner}/{repo.removesuffix('.git')}/compare/{old_rev_tag}...{new_rev_tag}"
```

Comment:
We probably shouldn't set diff_url if either the old or new rev tag is None:

```suggestion
            if old_rev_tag is not None and new_rev_tag is not None:
                package.diff_url = f"https://github.com/{owner}/{repo.removesuffix('.git')}/compare/{old_rev_tag}...{new_rev_tag}"
```